### PR TITLE
Move less mature extensions out of core repo

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #946 

**Proposed Changes:**

1. Update extension readme to talk about core vs 'community' extensions
2. Removed all non-core extensions (everything but view, eo, proj and scientific)
3. Updated examples and text to only refer to the core extensions (except in best practices)
4. Made list of community extensions (some links will be broken until we create the repos)

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [ ] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md)

### CODE REVIEW

**Comment by cholmes** [comment]
Ok, I believe this is ready for review, though the links to the repos not yet created will be broken. Should check those after all are created.

**Review by lossyrob — APPROVED** [review]

**Comment by lossyrob** [comment]
Reviewed, also created all the repos for stac-extensions so links should not be broken

**Review by matthewhanson — APPROVED** [review]

File: extensions/scientific/examples/item.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "stac_version": "1.0.0-beta.2",
   "stac_extensions": [
-    "file",
```
Comment (m-mohr) [inline extensions/scientific/examples/item.json:4]: The file:checksum is still in here.

**Review by m-mohr — APPROVED** [review]
Looks good, but there's one comment to solve.

**Pull request merged**